packages/components/src/context-tag/context-tag.tsx

```diff
@@ -1,93 +1,230 @@
-import { cloneElement, Fragment } from 'react'
-
-import { ChevronRightIcon } from '@status-im/icons'
-import { styled } from '@tamagui/core'
+import { ChevronRightIcon, PlayIcon } from '@status-im/icons'
+import { Stack, styled } from '@tamagui/core'
 import { View } from 'react-native'
 
 import { Avatar } from '../avatar'
 import { Text } from '../text'
 
-import type { AvatarProps } from '../avatar'
 import type { TextProps } from '../text'
 
-type ContextTagType =
-  | 'default'
-  | 'group'
-  | 'channel'
-  | 'community'
-  | 'token'
-  | 'network'
-  | 'account'
-  | 'collectible'
-  | 'address'
-  | 'icon'
-  | 'audio'
-
 type Props = {
   children?: React.ReactNode
-  src?: string
-  icon?: React.ReactElement
-  label: string | [string, string]
-  type?: ContextTagType
   size?: 24 | 32
   blur?: boolean
   outline?: boolean
-}
+} & (
+  | { type: 'default'; user: { name: string; src: string } }
+  | { type: 'account'; user: { name: string; emoji: string } }
+  | {
+      type: 'group'
+      group: {
+        name: string
+        icon: React.ReactElement
+      }
+    }
+  | { type: 'community'; community: { name: string; src: string } }
+  | {
+      type: 'channel'
+      channel: { communityName: string; src: string; name: string }
+    }
+  | { type: 'token'; token: { name: string; src: string } }
+  | { type: 'network'; network: { name: string; src: string } }
+  | { type: 'collectible'; collectible: { name: string; src: string } }
+  | { type: 'address'; address: string }
+  | { type: 'icon'; icon: React.ReactElement; label: string }
+  | { type: 'audio'; audioLength: string }
+)
 
 const textSizes: Record<NonNullable<Props['size']>, TextProps['size']> = {
   '32': 15,
   '24': 13,
 }
 
-const avatarSizes: Record<NonNullable<Props['size']>, AvatarProps['size']> = {
+const avatarSizes: Record<NonNullable<Props['size']>, 28 | 20> = {
   '32': 28,
   '24': 20,
 }
 
-const Label = ({ children, size }: { children: string; size: 24 | 32 }) => (
-  <Text size={textSizes[size]} weight="medium" color="$neutral-100">
+const Label = ({
+  children,
+  size,
+  type = 'default',
+}: {
+  children: string
+  size: 24 | 32
+  type?: 'default' | 'monospace'
+}) => (
+  <Text size={textSizes[size]} weight="medium" color="$neutral-100" type={type}>
     {children}
   </Text>
 )
 
 const ContextTag = (props: Props) => {
-  const {
-    src,
-    icon,
-    label,
-    // type = 'default', // this is commented because it's not being used
-    size = 24,
-    blur = false,
-    outline,
-  } = props
+  const { size = 24, blur = false, outline, type } = props
 
-  const hasImg = Boolean(src || icon)
+  const rounded = type === 'account' || type === 'collectible'
+  const leftShelf = type !== 'address'
```

Comment:
What exactly does "shelf" mean here?

Reply:
Basically if there will be something rendered on left side of the tag. Previously it was not just avatar, it could be icon or image. Can rename it to `containAvatar` now or something like that.

Reply:
I suggest `hasAvatar`.